src/app/legacy/containers/MostRead/Amp/index.jsx

```diff
@@ -35,6 +35,11 @@ const rankTranslationScript = (endpoint, service) => {
         if (!item.promo.headlines.shortHeadline) {
           item.promo.headlines.shortHeadline = item.promo.headlines.seoHeadline;
         }
+
+        if(!item.promo.locators.assetUri) {
+          item.promo.locators.assetUri = item.promo.locators.canonicalUrl;
+        }
```

Comment:
Is there any way we could check this logic in a unit test, as well as Cypress?

Reply:
So I was struggling to get data back in the unit tests. I think because the AMP scripts are not compatible with Jest. Tests seem to check against template tags (see attached).

This comment was already in the unit test file, which is why I used Cypress:

`  // Testing is done in cypress as jest dom does not support/run web workers which run the amp-scripts.`

Though let me know if I've missed another way to write unit tests!

![Screenshot 2023-02-16 at 15 01 27](https://user-images.githubusercontent.com/98817636/219401786-7f5f8bb2-9182-4736-bd0b-0ac9d4697aa0.png)

Reply:
No this is perfect, thanks for explaining. As long as we have code coverage (which we do!) I am happy. Thanks for fixing this!